spotlight-client/src/NarrativeLayout/NarrativeLayout.tsx

```diff
@@ -55,108 +56,62 @@ type NarrativeLayoutProps = {
 };
 
 const NarrativeLayout: React.FC<NarrativeLayoutProps> = ({ sections }) => {
-  const sectionsContainerRef = useRef() as React.MutableRefObject<HTMLDivElement>;
   const showSectionNavigation = useBreakpoint(true, ["mobile-", false]);
 
-  // automated scrolling is a special case of section visibility;
-  // this flag lets us suspend in-page navigation actions while it is in progress
-  // (it is a local variable rather than a piece of React state because
-  // the animation functions are outside the render loop and won't receive state updates)
-  let isScrolling = false;
-  const cancelAutoScroll = () => {
-    isScrolling = false;
-  };
-
-  const [activeSection, directlySetActiveSection] = useState(1);
-  // wrap the section state setter in a function that respects the flag
-  const setActiveSection = useCallback(
-    (sectionNumber: number) => {
-      if (!isScrolling) {
-        directlySetActiveSection(sectionNumber);
-      }
-    },
-    [isScrolling]
-  );
-
-  const [, setScrollSpring] = useSpring(() => ({
-    onFrame: (props: { top: number }) => {
-      if (isScrolling) window.scrollTo(0, props.top);
-    },
-    // set the flag while animation is in progress,
-    // and listen for user-initiated scrolling (which takes priority)
-    onRest: () => {
-      isScrolling = false;
-      window.removeEventListener("wheel", cancelAutoScroll);
-      window.removeEventListener("touchmove", cancelAutoScroll);
-    },
-    onStart: () => {
-      isScrolling = true;
-      window.addEventListener("wheel", cancelAutoScroll, { once: true });
-      window.addEventListener("touchmove", cancelAutoScroll, { once: true });
-    },
-    to: { top: window.pageYOffset },
-  }));
-
-  // updating the active section has two key side effects:
-  // 1. smoothly scrolling to the active section
-  // 2. updating the page URL so the section can be linked to directly
-  useEffect(() => {
-    let scrollDestination;
-    // scroll to the corresponding section by calculating its offset
-    const desiredSection = sectionsContainerRef.current.querySelector(
-      `#section${activeSection}`
-    );
-    if (desiredSection) {
-      scrollDestination =
-        window.pageYOffset + desiredSection.getBoundingClientRect().top;
-    }
-
-    // in practice this should always be defined, this is just type safety
-    if (scrollDestination !== undefined) {
-      setScrollSpring({
-        to: { top: scrollDestination - NAV_BAR_HEIGHT },
-        from: { top: window.pageYOffset },
-        reset: true,
-      });
-    }
-  }, [activeSection, sectionsContainerRef, setScrollSpring]);
+  const {
+    alwaysExpanded,
+    currentSectionNumber,
+    enableSnapping,
+    fixedHeightSections,
+    scrollToSection,
+    sectionsContainerRef,
+    getOnSectionExpanded,
+    onInViewChange,
```

Comment:
Wondering what you think about calling this variable `onSectionInViewChange`, might be easier to read later on without needing to checkout the note in the hook.